x-pack/docs/en/security/authentication/jwt-realm.asciidoc

```diff
@@ -0,0 +1,317 @@
+[role="xpack"]
+[[jwt-realm]]
+=== JWT authentication
+
+beta::[]
+
+{es} can be configured to trust JSON Web Tokens (JWTs) that are issued as an 
+authentication credential from an external service.
+
+When a JWT realm is used to authenticate with {es}, a distinction is made
+between the _client_ that is connecting to {es}, and the _user_ on whose behalf
+the request should run. The JWT identifies the user, and a separate credential
+is used to authenticate the client.
+
+A common scenario that uses JWTs is when an existing front-end application uses 
+OpenID Connect (OIDC) as an authentication method, and then accesses {es}
+on behalf of the authenticated user.
+
+TIP: If the front-end application does not exist, you can use the
+<<token-authentication-services>> instead.
+
+[[jwt-realm-oidc]]
+==== JWT uses OIDC workflows
+JWT authentication in {es} is derived from OIDC workflows, where users initiate
+contact with an OIDC Relying Party (RP) and are redirected to authenticate with
+an OIDC Provider (OP). Separate tokens are issued at different steps in the
+workflow to represent authentication, authorization, and identification.
+
+When users authenticate to the OP, they receive a _grant token_, which they
+present to the RP as proof of authentication. The RP then asks the OP to verify
+the  grant token and exchange it for two tokens: an _ID token_ and an
+_access token_.
```

Comment:
This is not necessarily true.
There are a lot of different OIDC flows, and the one described here is just one of them. The JWT realm supports them all, and I think that it may create confusion to the reader.

I'd rather use a more generic content, something like this:
```suggestion
JWT authentication in {es} is derived from OIDC workflows, where different tokens may be issued by an OIDC Provider (OP). One of these tokens is called the _ID token_ and uses the JWT format.
```

Reply:
I'll change this to read:
>JWT authentication in {es} is derived from OIDC workflows, where different tokens can be issued by an OIDC Provider (OP). One possible token is an _ID token_, which uses the JWT format.